### PR TITLE
Stop registering Kube cluster named after Teleport cluster

### PR DESCRIPTION
**Pull request by nklaassen**
Issue #6226

There are essentially 3 ways a Teleport Kubernetes service can be configured inside a Kubernetes cluster, listed below.

This PR only changes the first case where the new style Proxy Service is used (`kube_listen_addr` set, no `kubernetes:` subsection) inside of a Kubernetes cluster. Currently, in this case, the Proxy service will publish a Kubernetes cluster with the same name as the Teleport cluster. This is not desired - the Kubernetes cluster name should only be published by the `kubernetes_service`.

### 1. New Proxy Service style with Kubernetes Service
```
auth_service:
  cluster_name: peach

proxy_service:
  kube_listen_addr: 0.0.0.0:3026

kubernetes_service:
  enabled: yes
  listen_addr: 0.0.0.0:3027
  kube_cluster_name: banana
```

Current behaviour:
```sh
$ tsh kube ls
peach
banana
```
New behaviour:
```sh
$ tsh kube ls
banana
```
### 2. Legacy Proxy Service style without Kubernetes Service
```
auth_service:
  cluster_name: peach

proxy_service:
  kubernetes:
    enabled: yes
    listen_addr: 0.0.0.0:3026
```

Current behaviour (unchanged):
```sh
$ tsh kube ls
peach
```

### 3. Legacy Proxy Service style with Kubernetes Service
```
auth_service:
  cluster_name: peach

proxy_service:
  kubernetes:
    enabled: yes
    listen_addr: 0.0.0.0:3026

kubernetes_service:
  enabled: yes
  listen_addr: 0.0.0.0:3027
  kube_cluster_name: banana
```

Current behaviour (unchanged):
```sh
$ tsh kube ls
peach
banana
```

## Backwards Compatability
Anyone currently using the new style Proxy Service with `kube_listen_addr` set, inside of a Kubernetes cluster, will have the current listed Kubernetes cluster named after the teleport cluster "disappear". This may break automation that relies on that "cluster".

## Testing
- `configuration_test.go` and `auth_test.go` updated
- manually tested each configuration on a local Kubernetes cluster

### CODE REVIEW

File: lib/kube/proxy/forwarder.go [hunk 1]
```diff
@@ -93,10 +105,8 @@ type ForwarderConfig struct {
 	Context context.Context
 	// KubeconfigPath is a path to kubernetes configuration
 	KubeconfigPath string
-	// NewKubeService specifies whether to apply the additional kubernetes_service features:
-	// - parsing multiple kubeconfig entries
-	// - enforcing self permission check
-	NewKubeService bool
+	// KubeServiceType specifies which Teleport service type this forwarder is for
+	KubeServiceType
```
Comment (tcsc) [inline lib/kube/proxy/forwarder.go:109]: Does this need to be embedded?

File: lib/kube/proxy/forwarder.go [hunk 2]
```diff
@@ -62,6 +62,18 @@ import (
 	utilexec "k8s.io/client-go/util/exec"
 )
 
+// KubeServiceType specifies a Teleport service type which can proxy Kubernetes requests
+type KubeServiceType int
+
+const (
+	// KubeService is a Teleport kubernetes_service
```
Comment (tcsc) [inline lib/kube/proxy/forwarder.go:69]: I think these `KubeServiceType` values might need a bit more info about what they imply about the k8s service they apply to.

Comment (nklaassen) [reply lib/kube/proxy/forwarder.go]: I've expanded on these comments in the latest commit.

**Review by tcsc — APPROVED** [review]
Looks OK to me.

File: lib/kube/proxy/auth.go [hunk 3]
```diff
@@ -62,43 +62,59 @@ func TestOnlySkipSelfPermissionCheck(skip bool) {
...
 		// Also, remove all other contexts. Multiple kubeconfig entries are
 		// only supported for kubernetes_service.
-		cfg.Contexts = map[string]*rest.Config{
-			tpClusterName: cfg.Contexts[cfg.CurrentContext],
+		if currentContext, ok := cfg.Contexts[cfg.CurrentContext]; ok {
+			cfg.Contexts = map[string]*rest.Config{
+				tpClusterName: currentContext,
+			}
+		} else {
+			return nil, trace.BadParameter("No Kubernetes current-context found; Kubernetes proxy service requires either a valid kubeconfig_path with a current-context or to run inside of a kubernetes pod")
```
Comment (awly) [inline lib/kube/proxy/auth.go:117]:
```suggestion
			return nil, trace.BadParameter("no Kubernetes current-context found; Kubernetes proxy service requires either a valid kubeconfig_path with a current-context or to run inside of a Kubernetes pod")
```

File: lib/kube/proxy/forwarder.go [hunk 4]
```diff
@@ -156,7 +171,7 @@ func (f *ForwarderConfig) CheckAndSetDefaults() error {
 	if f.Component == "" {
 		f.Component = "kube_forwarder"
 	}
```
Comment (awly) [inline lib/kube/proxy/forwarder.go:173]: Validate that `KubeServiceType` is one of the 3 known values

**Review by awly — APPROVED** [review]

File: lib/kube/proxy/forwarder.go (hunk 4 above)
Comment (nklaassen) [reply lib/kube/proxy/forwarder.go:173]: done in latest commit

**Review by r0mant — APPROVED** [review]

**Comment by nklaassen** [comment]
Tagging @klizhentas for UX approval

**Comment by klizhentas** [comment]
@nklaassen LGTM after you update CHANGELOG.md with a breaking change notification similar to others (see examples in changelog) (so we don't forget)

**Review by klizhentas — APPROVED** [review]
LGTM assuming changelog update

**Comment by nklaassen** [comment]
> @nklaassen LGTM after you update CHANGELOG.md with a breaking change notification similar to others (see examples in changelog) (so we don't forget)

Thanks, I started a placeholder 7.0 section in CHANGELOG.md with a Breaking Changes section including this

**Pull request merged**